server/controllers/locationController.ts

```diff
@@ -0,0 +1,95 @@
+import { Request, Response } from 'express';
+import { Like } from 'typeorm';
+
+import { Location } from 'server/models/Location';
+
+export default {
+  async index(req: Request, res: Response) {
+    const { country_code, city, region, postal_code, address } = req.query;
+
+    const locations = await Location.find({
+      where: [
+        { country_code: Like(`%${country_code}%`) },
+        { city: Like(`%${city}%`) },
+        { region: Like(`%${region}%`) },
+        { postal_code: Like(`%${postal_code}%`) },
+        { address: Like(`%${address}%`) },
+      ],
+    });
+
+    res.json(locations);
+  },
+
+  async show(req: Request, res: Response) {
+    const { id } = req.params;
+
+    const location = await Location.findOne({ id: parseInt(id) });
+
+    if (location) {
+      res.json(location);
+    } else {
+      res.status(404).json({ error: "Can't find location" });
+    }
+  },
+
+  async create(req: Request, res: Response) {
+    const { country_code, city, region, postal_code, address } = req.body;
+
+    const location = new Location({
+      country_code,
+      city,
+      region,
+      postal_code,
+      address,
+    });
+
+    try {
+      await location.save();
+      res.status(201).json(location);
+    } catch (e) {
+      res.status(500).json({ error: e });
+    }
+  },
+  async update(req: Request, res: Response) {
+    const { id } = req.params;
+    const { country_code, city, region, postal_code, address } = req.body;
+
+    const location = await Location.findOne({ id: parseInt(id) });
+
+    if (location) {
```

Comment:
Instead of wrapping everything in this `if` block, it's cleaner to instead check `if (!location)` then return error, otherwise continue as normal.